src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -75,6 +75,10 @@ public static unsafe object RhNewArray(MethodTable* pEEType, int length)
         [RuntimeExport("RhBox")]
         public static unsafe object RhBox(MethodTable* pEEType, ref byte data)
         {
+            // Compatibility with CoreCLR, throw on a null reference to the unboxed data.
+            if (Unsafe.IsNullRef(ref target))
```

Comment:
```suggestion
            // A null can be passed for boxing of a null ref.
            if (Unsafe.IsNullRef(ref data))
```
Match comment in CoreCLR impl